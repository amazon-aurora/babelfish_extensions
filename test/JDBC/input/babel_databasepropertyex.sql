-- test databasepropertyex() function
-- invalid db name, should return NULL
select databasepropertyex(N'invalid_dbname',N'Edition');
GO
-- invalid property name, should return NULL
select databasepropertyex(N'template1',N'invalid_property');
GO
-- test return type
select pg_typeof(databasepropertyex(N'template1',N'invalid_property'));
GO
-- test supported properties
select databasepropertyex(N'template1',N'Collation');
GO
select databasepropertyex(N'template1',N'IsInStandBy');
GO
select databasepropertyex(N'template1',N'IsAutoClose');
GO
select databasepropertyex(N'template1',N'IsAutoCreateStatistics');
GO
select 'true' where databasepropertyex(N'template1',N'IsTornPageDetectionEnabled') >= 0
GO
select databasepropertyex(N'template1',N'Updateability');
GO
select databasepropertyex(N'template1',N'Status');
GO
SELECT (case when charindex(cast(databasepropertyex(N'template1',N'Version') as nvarchar), version()) > 0 then 'true' else 'false' end) result
GO
-- test unsupported properties(expect return NULL or 0)
select databasepropertyex(N'template1',N'IsArithmeticAbortEnabled');
GO
select databasepropertyex(N'template1',N'IsAutoShrink');
GO
select databasepropertyex(N'template1',N'MaxSizeInBytes');
GO
