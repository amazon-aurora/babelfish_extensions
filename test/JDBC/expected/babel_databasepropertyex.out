-- test databasepropertyex() function
-- invalid db name, should return NULL
select databasepropertyex(N'invalid_dbname',N'Edition');
GO
~~START~~
sql_variant
<NULL>
~~END~~

-- invalid property name, should return NULL
select databasepropertyex(N'template1',N'invalid_property');
GO
~~START~~
sql_variant
<NULL>
~~END~~

-- test return type
select pg_typeof(databasepropertyex(N'template1',N'invalid_property'));
GO
~~ERROR (Code: 33557097)~~

~~ERROR (Message: data type regtype is not supported yet)~~

-- test supported properties
select databasepropertyex(N'template1',N'Collation');
GO
~~START~~
sql_variant
sql_latin1_general_cp1_ci_as
~~END~~

select databasepropertyex(N'template1',N'IsInStandBy');
GO
~~START~~
sql_variant
0
~~END~~

select databasepropertyex(N'template1',N'IsAutoClose');
GO
~~START~~
sql_variant
0
~~END~~

select databasepropertyex(N'template1',N'IsAutoCreateStatistics');
GO
~~START~~
sql_variant
1
~~END~~

select 'true' where databasepropertyex(N'template1',N'IsTornPageDetectionEnabled') >= 0
GO
~~START~~
varchar
true
~~END~~

select databasepropertyex(N'template1',N'Updateability');
GO
~~START~~
sql_variant
READ_WRITE
~~END~~

select databasepropertyex(N'template1',N'Status');
GO
~~START~~
sql_variant
ONLINE
~~END~~

SELECT (case when charindex(cast(databasepropertyex(N'template1',N'Version') as nvarchar), version()) > 0 then 'true' else 'false' end) result
GO
~~START~~
text
true
~~END~~

-- test unsupported properties(expect return NULL or 0)
select databasepropertyex(N'template1',N'IsArithmeticAbortEnabled');
GO
~~START~~
sql_variant
0
~~END~~

select databasepropertyex(N'template1',N'IsAutoShrink');
GO
~~START~~
sql_variant
0
~~END~~

select databasepropertyex(N'template1',N'MaxSizeInBytes');
GO
~~START~~
sql_variant
<NULL>
~~END~~

